### DISCUSSION TITLE
docker secure login

### DISCUSSION THREAD

**Discussion by electriquo** [semantic-release-plus/semantic-release-plus#125] [Q&A] [answered]
https://github.com/semantic-release-plus/semantic-release-plus/blob/60792c3e1e066ac37186de5477f559decad59bd8/packages/plugins/docker/src/lib/docker-utils.ts#L13-L15

current implementation causes the following warning message

> WARNING! Using -*** the CLI is insecure. Use --password-stdin.

reading through [Provide a password using STDIN](https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin) docker documentation:
> To run the docker login command non-interactively, you can set the --password-stdin flag to provide a password through STDIN. Using STDIN prevents the password from ending up in the shell’s history, or log-files.

the following is much more secure

```shell
echo $DOCKER_PASSWORD | docker login registry --username $DOCKER_USERNAME --password-stdin
```

any reason why the more secure method was not used in the first place?

**Comment by JoA-MoS** [comment, marked as answer]
This was fixed a bit ago, can you try the latest version, if you are coming from 2.x.x there are some breaking changes.
